tests/init_config_test.rs

```diff
@@ -0,0 +1,57 @@
+use scroll_proving_sdk::{config::Config, prover::CircuitType};
+use sindri_scroll_sdk::prover::CloudProver;
+use std::env::set_var;
+
+// Ensures that configuration file loading does not require environment variables
+#[tokio::test]
+async fn test_config_without_envs() {
+    let default_config_path = "tests/test_data/default_config.json";
+    let cfg: Config = Config::from_file_and_env(default_config_path.to_string())
+        .expect("Issue loading test configuration file");
+
+    // Check a few values for consistency with test_data file
+    assert_eq!(cfg.prover_name_prefix, "sindri_");
+    assert_eq!(cfg.prover.circuit_type, CircuitType::Bundle);
+}
+
+// Ensures that various environment variable overrides are successful
+#[test]
+fn test_config_with_envs() {
+    // Establish some environment variable overrides
+
+    set_var("N_WORKERS", "10");
+    set_var("COORDINATOR_BASE_URL", "my-special-coordinator");
+
+    let default_config_path = "tests/test_data/default_config.json";
+    let cfg: Config = Config::from_file_and_env(default_config_path.to_string())
+        .expect("Issue loading test configuration file");
+
+    // Ensure override was successful
+    assert_eq!(
+        cfg.prover.n_workers, 10_usize,
```

Comment:
This assertion becomes useless if the `n_workers` value in test_data/default_config.json happens to be changed to 10. Perhaps verify the expected value from the default file in `test_config_without_envs()` as a sanity check?

Reply:
I committed your suggestion in https://github.com/Sindri-Labs/sindri-scroll-sdk/pull/28/commits/b33985c3916fc111e83c79adad9d8576462e28a2 which added the suggested check:
`assert_eq!(cfg.prover.n_workers, 1_usize);`

Reply:
The suggestion uncovered an unfortunate side effect of the `std::env::set_var` method - these are not local to the function where you call them.  The added assertion caused `test_config_without_envs` to fail because `test_config_with_envs` sets the number of workers and then does not unset it.

So commit https://github.com/Sindri-Labs/sindri-scroll-sdk/pull/28/commits/ffcea40a209c8b35418150125c618e4c0e59c588 includes the use of `std::end::remove_var` to get rid of those effects

Reply:
My journey was not over when I posted the above.  A few failed CI checks later, I figured out that the parallel execution of these tests is not predictable and the `remove_var` solution only works a fraction of the time.  Generally `test_config_without_envs` would fail because it is running at the same time as `test_config_with_envs`.

Research led me to this lightweight crate [temp-env](https://crates.io/crates/temp-env) which solves this exact problem (setting environment variables within independent testing threads).  I had to wrap both our `test_config_with_envs` and `test_config_without_envs` tests in their own closures that do not execute at the same time.

Reply:
`temp_env` nice find for a tricky test failure (unpredictable execution order).